src/include/duckdb/execution/operator/persistent/csv_scanner/csv_buffer.hpp

```diff
@@ -0,0 +1,102 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/execution/operator/persistent/csv_scanner/csv_buffer.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/constants.hpp"
+#include "duckdb/execution/operator/persistent/csv_scanner/csv_file_handle.hpp"
+#include "duckdb/storage/buffer_manager.hpp"
+#include "duckdb/storage/block_manager.hpp"
+#include "duckdb/storage/buffer/block_handle.hpp"
+
+namespace duckdb {
+static constexpr idx_t CSV_BUFFER_SIZE = 32000000; // 32MB
+//! In case the file has a size < 32MB, we will use this size instead
+//! This is to avoid mallocing a lot of memory for a small file
+//! And if it's a compressed file we can't use the actual size of the file
+static constexpr idx_t CSV_MINIMUM_BUFFER_SIZE = 10000000; // 10MB
```

Comment:
Do they need to be global? Or can we move them into the `CSVBuffer` class? You only use it in `csv_buffer_manager.cpp` and `csv_reader_options.hpp`, and I think we usually keep our `constexpr` inside the context of structs or classes.